### ISSUE TITLE
test_no_warnings broken on newer stable Rust

### ISSUE THREAD

**Issue by luser** [brendanzab/gl-rs#416]
My PR (https://github.com/brendanzab/gl-rs/pull/415) failed Travis, and per https://github.com/brendanzab/gl-rs/pull/411#issuecomment-270483188 the last Travis build on master was almost 6 months ago. January 14th, 2017, to be specific: https://github.com/brendanzab/gl-rs/commit/a76dce48ab5374a29c4112f8d853a7c324d4da44

It looks like this test was broken in Rust 1.15. I installed a few older releases via rustup and worked backwards until it passed, which it did with Rust 1.14:

<details><summary>Rust 1.14 results</summary><p>

```
luser@eye7:/build/gl-rs$ cargo +1.14.0 test --manifest-path gl_tests/test_no_warnings/Cargo.toml 
   Compiling khronos_api v1.0.0 (file:///build/gl-rs/khronos_api)
   Compiling bitflags v0.7.0
   Compiling log v0.3.7
   Compiling xml-rs v0.3.6
   Compiling gl_generator v0.5.2 (file:///build/gl-rs/gl_generator)
   Compiling test_no_warnings v0.0.0 (file:///build/gl-rs/gl_tests/test_no_warnings)
    Finished debug [unoptimized + debuginfo] target(s) in 66.56 secs
     Running gl_tests/test_no_warnings/target/debug/deps/test_no_warnings-54cea259b0a8ed88

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured

   Doc-tests test_no_warnings

running 12 tests
test egl_debug_struct::Egl::load_with_0 ... ignored
test egl_struct::Egl::load_with_0 ... ignored
test gl_debug_struct::Gl::load_with_0 ... ignored
test gl_struct::Gl::load_with_0 ... ignored
test gles1_debug_struct::Gles1::load_with_0 ... ignored
test gles1_struct::Gles1::load_with_0 ... ignored
test gles2_debug_struct::Gles2::load_with_0 ... ignored
test gles2_struct::Gles2::load_with_0 ... ignored
test glx_debug_struct::Glx::load_with_0 ... ignored
test glx_struct::Glx::load_with_0 ... ignored
test wgl_debug_struct::Wgl::load_with_0 ... ignored
test wgl_struct::Wgl::load_with_0 ... ignored

test result: ok. 0 passed; 0 failed; 12 ignored; 0 measured
```

</p></details>

and it fails with Rust 1.15:

<details><summary>Rust 1.15 results</summary><p>

```
luser@eye7:/build/gl-rs$ cargo +1.15.0 test --manifest-path gl_tests/test_no_warnings/Cargo.toml 
   Compiling test_no_warnings v0.0.0 (file:///build/gl-rs/gl_tests/test_no_warnings)
error: type alias is never used: `khronos_uint64_t`
     --> /build/gl-rs/gl_tests/test_no_warnings/target/debug/build/test_no_warnings-77e6a93acbd794f4/out/test_no_warnings.rs:81338:9
      |
81338 |         pub type khronos_uint64_t = u64;
      |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      |
note: lint level defined here
     --> gl_tests/test_no_warnings/lib.rs:16:9
      |
16    | #![deny(warnings)]
      |         ^^^^^^^^

error: type alias is never used: `khronos_ssize_t`
     --> /build/gl-rs/gl_tests/test_no_warnings/target/debug/build/test_no_warnings-77e6a93acbd794f4/out/test_no_warnings.rs:81339:9
      |
81339 |         pub type khronos_ssize_t = isize;
      |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: type alias is never used: `NativeDisplayType`
     --> /build/gl-rs/gl_tests/test_no_warnings/target/debug/build/test_no_warnings-77e6a93acbd794f4/out/test_no_warnings.rs:81344:9
      |
81344 |         pub type NativeDisplayType = *const raw::c_void;
      |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: type alias is never used: `NativePixmapType`
     --> /build/gl-rs/gl_tests/test_no_warnings/target/debug/build/test_no_warnings-77e6a93acbd794f4/out/test_no_warnings.rs:81345:9
      |
81345 |         pub type NativePixmapType = *const raw::c_void;
      |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: type alias is never used: `NativeWindowType`
     --> /build/gl-rs/gl_tests/test_no_warnings/target/debug/build/test_no_warnings-77e6a93acbd794f4/out/test_no_warnings.rs:81346:9
      |
81346 |         pub type NativeWindowType = *const raw::c_void;
      |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: type alias is never used: `khronos_utime_nanoseconds_t`
     --> /build/gl-rs/gl_tests/test_no_warnings/target/debug/build/test_no_warnings-77e6a93acbd794f4/out/test_no_warnings.rs:82963:9
      |
82963 |         pub type khronos_utime_nanoseconds_t = raw::c_int;
      |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: type alias is never used: `khronos_uint64_t`
     --> /build/gl-rs/gl_tests/test_no_warnings/target/debug/build/test_no_warnings-77e6a93acbd794f4/out/test_no_warnings.rs:82964:9
      |
82964 |         pub type khronos_uint64_t = u64;
      |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: type alias is never used: `khronos_ssize_t`
     --> /build/gl-rs/gl_tests/test_no_warnings/target/debug/build/test_no_warnings-77e6a93acbd794f4/out/test_no_warnings.rs:82965:9
      |
82965 |         pub type khronos_ssize_t = isize;
      |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: type alias is never used: `EGLNativePixmapType`
     --> /build/gl-rs/gl_tests/test_no_warnings/target/debug/build/test_no_warnings-77e6a93acbd794f4/out/test_no_warnings.rs:82967:9
      |
82967 |         pub type EGLNativePixmapType = *const raw::c_void;
      |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: type alias is never used: `EGLNativeWindowType`
     --> /build/gl-rs/gl_tests/test_no_warnings/target/debug/build/test_no_warnings-77e6a93acbd794f4/out/test_no_warnings.rs:82968:9
      |
82968 |         pub type EGLNativeWindowType = *const raw::c_void;
      |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: type alias is never used: `NativeDisplayType`
     --> /build/gl-rs/gl_tests/test_no_warnings/target/debug/build/test_no_warnings-77e6a93acbd794f4/out/test_no_warnings.rs:82970:9
      |
82970 |         pub type NativeDisplayType = *const raw::c_void;
      |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: type alias is never used: `NativePixmapType`
     --> /build/gl-rs/gl_tests/test_no_warnings/target/debug/build/test_no_warnings-77e6a93acbd794f4/out/test_no_warnings.rs:82971:9
      |
82971 |         pub type NativePixmapType = *const raw::c_void;
      |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: type alias is never used: `NativeWindowType`
     --> /build/gl-rs/gl_tests/test_no_warnings/target/debug/build/test_no_warnings-77e6a93acbd794f4/out/test_no_warnings.rs:82972:9
      |
82972 |         pub type NativeWindowType = *const raw::c_void;
      |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: type alias is never used: `khronos_uint64_t`
     --> /build/gl-rs/gl_tests/test_no_warnings/target/debug/build/test_no_warnings-77e6a93acbd794f4/out/test_no_warnings.rs:83388:9
      |
83388 |         pub type khronos_uint64_t = u64;
      |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: type alias is never used: `khronos_ssize_t`
     --> /build/gl-rs/gl_tests/test_no_warnings/target/debug/build/test_no_warnings-77e6a93acbd794f4/out/test_no_warnings.rs:83389:9
      |
83389 |         pub type khronos_ssize_t = isize;
      |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: type alias is never used: `NativeDisplayType`
     --> /build/gl-rs/gl_tests/test_no_warnings/target/debug/build/test_no_warnings-77e6a93acbd794f4/out/test_no_warnings.rs:83394:9
      |
83394 |         pub type NativeDisplayType = *const raw::c_void;
      |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: type alias is never used: `NativePixmapType`
     --> /build/gl-rs/gl_tests/test_no_warnings/target/debug/build/test_no_warnings-77e6a93acbd794f4/out/test_no_warnings.rs:83395:9
      |
83395 |         pub type NativePixmapType = *const raw::c_void;
      |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: type alias is never used: `NativeWindowType`
     --> /build/gl-rs/gl_tests/test_no_warnings/target/debug/build/test_no_warnings-77e6a93acbd794f4/out/test_no_warnings.rs:83396:9
      |
83396 |         pub type NativeWindowType = *const raw::c_void;
      |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: type alias is never used: `khronos_uint64_t`
     --> /build/gl-rs/gl_tests/test_no_warnings/target/debug/build/test_no_warnings-77e6a93acbd794f4/out/test_no_warnings.rs:83973:9
      |
83973 |         pub type khronos_uint64_t = u64;
      |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: type alias is never used: `khronos_ssize_t`
     --> /build/gl-rs/gl_tests/test_no_warnings/target/debug/build/test_no_warnings-77e6a93acbd794f4/out/test_no_warnings.rs:83974:9
      |
83974 |         pub type khronos_ssize_t = isize;
      |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: type alias is never used: `NativeDisplayType`
     --> /build/gl-rs/gl_tests/test_no_warnings/target/debug/build/test_no_warnings-77e6a93acbd794f4/out/test_no_warnings.rs:83979:9
      |
83979 |         pub type NativeDisplayType = *const raw::c_void;
      |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: type alias is never used: `NativePixmapType`
     --> /build/gl-rs/gl_tests/test_no_warnings/target/debug/build/test_no_warnings-77e6a93acbd794f4/out/test_no_warnings.rs:83980:9
      |
83980 |         pub type NativePixmapType = *const raw::c_void;
      |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: type alias is never used: `NativeWindowType`
     --> /build/gl-rs/gl_tests/test_no_warnings/target/debug/build/test_no_warnings-77e6a93acbd794f4/out/test_no_warnings.rs:83981:9
      |
83981 |         pub type NativeWindowType = *const raw::c_void;
      |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: type alias is never used: `khronos_uint64_t`
     --> /build/gl-rs/gl_tests/test_no_warnings/target/debug/build/test_no_warnings-77e6a93acbd794f4/out/test_no_warnings.rs:84672:9
      |
84672 |         pub type khronos_uint64_t = u64;
      |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: type alias is never used: `khronos_ssize_t`
     --> /build/gl-rs/gl_tests/test_no_warnings/target/debug/build/test_no_warnings-77e6a93acbd794f4/out/test_no_warnings.rs:84673:9
      |
84673 |         pub type khronos_ssize_t = isize;
      |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: type alias is never used: `NativeDisplayType`
     --> /build/gl-rs/gl_tests/test_no_warnings/target/debug/build/test_no_warnings-77e6a93acbd794f4/out/test_no_warnings.rs:84678:9
      |
84678 |         pub type NativeDisplayType = *const raw::c_void;
      |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: type alias is never used: `NativePixmapType`
     --> /build/gl-rs/gl_tests/test_no_warnings/target/debug/build/test_no_warnings-77e6a93acbd794f4/out/test_no_warnings.rs:84679:9
      |
84679 |         pub type NativePixmapType = *const raw::c_void;
      |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: type alias is never used: `NativeWindowType`
     --> /build/gl-rs/gl_tests/test_no_warnings/target/debug/build/test_no_warnings-77e6a93acbd794f4/out/test_no_warnings.rs:84680:9
      |
84680 |         pub type NativeWindowType = *const raw::c_void;
      |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: type alias is never used: `khronos_uint64_t`
     --> /build/gl-rs/gl_tests/test_no_warnings/target/debug/build/test_no_warnings-77e6a93acbd794f4/out/test_no_warnings.rs:81338:9
      |
81338 |         pub type khronos_uint64_t = u64;
      |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
      |
note: lint level defined here
     --> gl_tests/test_no_warnings/lib.rs:16:9
      |
16    | #![deny(warnings)]
      |         ^^^^^^^^

error: type alias is never used: `khronos_ssize_t`
     --> /build/gl-rs/gl_tests/test_no_warnings/target/debug/build/test_no_warnings-77e6a93acbd794f4/out/test_no_warnings.rs:81339:9
      |
81339 |         pub type khronos_ssize_t = isize;
      |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: type alias is never used: `NativeDisplayType`
     --> /build/gl-rs/gl_tests/test_no_warnings/target/debug/build/test_no_warnings-77e6a93acbd794f4/out/test_no_warnings.rs:81344:9
      |
81344 |         pub type NativeDisplayType = *const raw::c_void;
      |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: type alias is never used: `NativePixmapType`
     --> /build/gl-rs/gl_tests/test_no_warnings/target/debug/build/test_no_warnings-77e6a93acbd794f4/out/test_no_warnings.rs:81345:9
      |
81345 |         pub type NativePixmapType = *const raw::c_void;
      |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: type alias is never used: `NativeWindowType`
     --> /build/gl-rs/gl_tests/test_no_warnings/target/debug/build/test_no_warnings-77e6a93acbd794f4/out/test_no_warnings.rs:81346:9
      |
81346 |         pub type NativeWindowType = *const raw::c_void;
      |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: type alias is never used: `khronos_utime_nanoseconds_t`
     --> /build/gl-rs/gl_tests/test_no_warnings/target/debug/build/test_no_warnings-77e6a93acbd794f4/out/test_no_warnings.rs:82963:9
      |
82963 |         pub type khronos_utime_nanoseconds_t = raw::c_int;
      |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: type alias is never used: `khronos_uint64_t`
     --> /build/gl-rs/gl_tests/test_no_warnings/target/debug/build/test_no_warnings-77e6a93acbd794f4/out/test_no_warnings.rs:82964:9
      |
82964 |         pub type khronos_uint64_t = u64;
      |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: type alias is never used: `khronos_ssize_t`
     --> /build/gl-rs/gl_tests/test_no_warnings/target/debug/build/test_no_warnings-77e6a93acbd794f4/out/test_no_warnings.rs:82965:9
      |
82965 |         pub type khronos_ssize_t = isize;
      |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: type alias is never used: `EGLNativePixmapType`
     --> /build/gl-rs/gl_tests/test_no_warnings/target/debug/build/test_no_warnings-77e6a93acbd794f4/out/test_no_warnings.rs:82967:9
      |
82967 |         pub type EGLNativePixmapType = *const raw::c_void;
      |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: type alias is never used: `EGLNativeWindowType`
     --> /build/gl-rs/gl_tests/test_no_warnings/target/debug/build/test_no_warnings-77e6a93acbd794f4/out/test_no_warnings.rs:82968:9
      |
82968 |         pub type EGLNativeWindowType = *const raw::c_void;
      |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: type alias is never used: `NativeDisplayType`
     --> /build/gl-rs/gl_tests/test_no_warnings/target/debug/build/test_no_warnings-77e6a93acbd794f4/out/test_no_warnings.rs:82970:9
      |
82970 |         pub type NativeDisplayType = *const raw::c_void;
      |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: type alias is never used: `NativePixmapType`
     --> /build/gl-rs/gl_tests/test_no_warnings/target/debug/build/test_no_warnings-77e6a93acbd794f4/out/test_no_warnings.rs:82971:9
      |
82971 |         pub type NativePixmapType = *const raw::c_void;
      |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: type alias is never used: `NativeWindowType`
     --> /build/gl-rs/gl_tests/test_no_warnings/target/debug/build/test_no_warnings-77e6a93acbd794f4/out/test_no_warnings.rs:82972:9
      |
82972 |         pub type NativeWindowType = *const raw::c_void;
      |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: type alias is never used: `khronos_uint64_t`
     --> /build/gl-rs/gl_tests/test_no_warnings/target/debug/build/test_no_warnings-77e6a93acbd794f4/out/test_no_warnings.rs:83388:9
      |
83388 |         pub type khronos_uint64_t = u64;
      |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: type alias is never used: `khronos_ssize_t`
     --> /build/gl-rs/gl_tests/test_no_warnings/target/debug/build/test_no_warnings-77e6a93acbd794f4/out/test_no_warnings.rs:83389:9
      |
83389 |         pub type khronos_ssize_t = isize;
      |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: type alias is never used: `NativeDisplayType`
     --> /build/gl-rs/gl_tests/test_no_warnings/target/debug/build/test_no_warnings-77e6a93acbd794f4/out/test_no_warnings.rs:83394:9
      |
83394 |         pub type NativeDisplayType = *const raw::c_void;
      |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: type alias is never used: `NativePixmapType`
     --> /build/gl-rs/gl_tests/test_no_warnings/target/debug/build/test_no_warnings-77e6a93acbd794f4/out/test_no_warnings.rs:83395:9
      |
83395 |         pub type NativePixmapType = *const raw::c_void;
      |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: type alias is never used: `NativeWindowType`
     --> /build/gl-rs/gl_tests/test_no_warnings/target/debug/build/test_no_warnings-77e6a93acbd794f4/out/test_no_warnings.rs:83396:9
      |
83396 |         pub type NativeWindowType = *const raw::c_void;
      |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: type alias is never used: `khronos_uint64_t`
     --> /build/gl-rs/gl_tests/test_no_warnings/target/debug/build/test_no_warnings-77e6a93acbd794f4/out/test_no_warnings.rs:83973:9
      |
83973 |         pub type khronos_uint64_t = u64;
      |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: type alias is never used: `khronos_ssize_t`
     --> /build/gl-rs/gl_tests/test_no_warnings/target/debug/build/test_no_warnings-77e6a93acbd794f4/out/test_no_warnings.rs:83974:9
      |
83974 |         pub type khronos_ssize_t = isize;
      |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: type alias is never used: `NativeDisplayType`
     --> /build/gl-rs/gl_tests/test_no_warnings/target/debug/build/test_no_warnings-77e6a93acbd794f4/out/test_no_warnings.rs:83979:9
      |
83979 |         pub type NativeDisplayType = *const raw::c_void;
      |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: type alias is never used: `NativePixmapType`
     --> /build/gl-rs/gl_tests/test_no_warnings/target/debug/build/test_no_warnings-77e6a93acbd794f4/out/test_no_warnings.rs:83980:9
      |
83980 |         pub type NativePixmapType = *const raw::c_void;
      |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: type alias is never used: `NativeWindowType`
     --> /build/gl-rs/gl_tests/test_no_warnings/target/debug/build/test_no_warnings-77e6a93acbd794f4/out/test_no_warnings.rs:83981:9
      |
83981 |         pub type NativeWindowType = *const raw::c_void;
      |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: type alias is never used: `khronos_uint64_t`
     --> /build/gl-rs/gl_tests/test_no_warnings/target/debug/build/test_no_warnings-77e6a93acbd794f4/out/test_no_warnings.rs:84672:9
      |
84672 |         pub type khronos_uint64_t = u64;
      |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: type alias is never used: `khronos_ssize_t`
     --> /build/gl-rs/gl_tests/test_no_warnings/target/debug/build/test_no_warnings-77e6a93acbd794f4/out/test_no_warnings.rs:84673:9
      |
84673 |         pub type khronos_ssize_t = isize;
      |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: type alias is never used: `NativeDisplayType`
     --> /build/gl-rs/gl_tests/test_no_warnings/target/debug/build/test_no_warnings-77e6a93acbd794f4/out/test_no_warnings.rs:84678:9
      |
84678 |         pub type NativeDisplayType = *const raw::c_void;
      |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: type alias is never used: `NativePixmapType`
     --> /build/gl-rs/gl_tests/test_no_warnings/target/debug/build/test_no_warnings-77e6a93acbd794f4/out/test_no_warnings.rs:84679:9
      |
84679 |         pub type NativePixmapType = *const raw::c_void;
      |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: type alias is never used: `NativeWindowType`
     --> /build/gl-rs/gl_tests/test_no_warnings/target/debug/build/test_no_warnings-77e6a93acbd794f4/out/test_no_warnings.rs:84680:9
      |
84680 |         pub type NativeWindowType = *const raw::c_void;
      |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 28 previous errors

Build failed, waiting for other jobs to finish...
error: aborting due to 28 previous errors

error: Could not compile `test_no_warnings`.

To learn more, run the command again with --verbose.
```

</p></details>

Rust 1.15 [was released Febuary 2nd, 2017](https://blog.rust-lang.org/2017/02/02/Rust-1.15.html), so the timing lines up as well.

**Comment by luser** [comment]
Nothing in [the release notes](https://github.com/rust-lang/rust/blob/master/RELEASES.md#version-1150-2017-02-02) stands out.

**Comment by luser** [comment]
I suspect this changed in https://github.com/rust-lang/rust/pull/29973 .

**Comment by brendanzab** [comment]
Added fixes to a bunch of issues in #417. Thanks for giving me a kick to get back to some housework!
